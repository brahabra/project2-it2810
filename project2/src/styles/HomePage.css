/* Css for Home page (lots of the styling happens in styles.tsx, because we used material ui.
material ui components cannot directly be styled in css.) */

.homeContainer {
    display: flex;
    flex-direction: column;
    gap: 2rem;
    height: 700px;
    justify-content: center;
    align-items: center;
}

.homepage-logo {
    width: 200px;
}
