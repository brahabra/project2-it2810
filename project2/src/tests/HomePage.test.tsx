import * as React from 'react'
import HomePage from '../pages/HomePage'
import {render, cleanup, screen} from "@testing-library/react"
import userEvent from '@testing-library/user-event';
import { wait } from '@testing-library/user-event/dist/utils';

const ReactTestRenderer = require('react-test-renderer');

//afterEach(cleanup)

it('renders correctly', () => {
  const tree = ReactTestRenderer.create(<HomePage />).toJSON()
  expect(tree).toMatchSnapshot()
})
/*
it('button', async () => {
  render(<HomePage/>)
  
  userEvent.click(screen.getByRole('button', {
    name: /submit/i
  }))
  await wait();
})
*/