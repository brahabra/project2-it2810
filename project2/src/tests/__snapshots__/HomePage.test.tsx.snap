// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <img
    alt="GitLab Logo"
    className="homepage-logo"
    src="test-file-stub"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="ProjectID"
      type="number"
      value={NaN}
    />
  </div>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Project Token"
      type="text"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Submit
  </button>
  <div
    className="feedbackText"
  >
     
     
  </div>
</div>
`;
